src/logging/logging.module.ts

```diff
@@ -0,0 +1,48 @@
+import { Global, Module } from '@nestjs/common';
+import { LoggingService } from './logging.interface';
+import { RequestScopedLoggingService } from './logging.service';
+import * as winston from 'winston';
+import * as Transport from 'winston-transport';
+
+/**
+ * Provides a new instance of a Winston logger using the provided {@link transports}
+ *
+ * @param transports - the logger transports to be used in the winston instance
+ */
+function winstonFactory(transports: Transport[] | Transport): winston.Logger {
+  return winston.createLogger({ transports: transports });
+}
+
+const LoggerTransports = Symbol('LoggerTransports');
+
+/**
+ * Factory which provides a collection of transports to be used by the
+ * logger instance
+ */
+function winstonTransportsFactory(): Transport[] | Transport {
+  return new winston.transports.Console({
+    level: 'debug',
+    format: winston.format.json(),
+  });
+}
+
+/**
+ * Module for logging messages throughout the application.
+ *
+ * Provides the RequestScopedLoggingService which logs the current time and request ID with every message.
+ */
+@Global()
+@Module({
+  providers: [
+    { provide: LoggingService, useClass: RequestScopedLoggingService },
+    { provide: LoggerTransports, useFactory: winstonTransportsFactory },
+    {
+      // Using a symbol or a string as a provider key doesn't work when using factories
```

Comment:
This comment is not 100% correct but I have more context since I wrote it 😅 

`string` works as a value if provided directly as you see it in the `provide:`. However if you extract the `string` to a `const` or `let` it stops working. I didn't have time to debug this (maybe the `const` needs to be exported due to how the modules are linked and built).

Reply:
I managed to use `Symbol` by exporting it from a separate file (as [indicated in the docs](https://docs.nestjs.com/fundamentals/dynamic-modules#module-configuration)) and importing here and into `LoggingService`. Not sure I'm happy with having a separate file just exporting a `Symbol`, do you find this setup valuable? Otherwise we can get back to using a plain string.

Here is the change: [c73623d](https://github.com/safe-global/safe-client-gateway-nest/pull/310/commits/c73623dec1bd6d2c7798b2608897c25e75a3c3a9)